Weights/include/CGAL/Weights/cotangent_weights.h

```diff
@@ -167,6 +167,331 @@ namespace Weights {
     const GeomTraits traits;
     return cotangent_weight(t, r, p, q, traits);
   }
+
+  // Undocumented cotangent weight class.
+  // Its constructor takes a polygon mesh and a vertex to point map
+  // and its operator() is defined based on the halfedge_descriptor only.
+  // This version is currently used in:
+  // Polygon_mesh_processing -> curvature_flow_impl.h
+  template<
+  typename PolygonMesh,
+  typename VertexPointMap = typename boost::property_map<PolygonMesh, CGAL::vertex_point_t>::type>
+  class Edge_cotangent_weight {
+
+    using GeomTraits = typename CGAL::Kernel_traits<
+        typename boost::property_traits<VertexPointMap>::value_type>::type;
+    using FT = typename GeomTraits::FT;
+
+    const PolygonMesh& m_pmesh;
+    const VertexPointMap m_pmap;
+    const GeomTraits m_traits;
+
+  public:
+    using vertex_descriptor = typename boost::graph_traits<PolygonMesh>::vertex_descriptor;
+    using halfedge_descriptor = typename boost::graph_traits<PolygonMesh>::halfedge_descriptor;
+
+    Edge_cotangent_weight(const PolygonMesh& pmesh, const VertexPointMap pmap) :
+    m_pmesh(pmesh), m_pmap(pmap), m_traits() { }
+
+    FT operator()(const halfedge_descriptor he) const {
+
+      FT weight = FT(0);
+      if (is_border_edge(he, m_pmesh)) {
+        const auto h1 = next(he, m_pmesh);
+
+        const auto v0 = target(he, m_pmesh);
+        const auto v1 = source(he, m_pmesh);
+        const auto v2 = target(h1, m_pmesh);
+
+        const auto& p0 = get(m_pmap, v0);
```

Comment:
@sloriot correct me if I am wrong, but does that not need systematically the use of `property_traits<>::reference`?

Reply:
Not necessarily, the lifetime of a temporary can be extended locally. So it is fine. I don't like too much the auto but that's a matter of taste I guess.